controlsfx/src/test/java/org/controlsfx/control/tableview2/TableView2Test.java

```diff
@@ -0,0 +1,303 @@
+                            if (index < 3) {
+                                fixed.add(column);
+                            }
+                            final Callback<TableColumn<RowItem, String>, TableCell<RowItem, String>> cellFactory = column.getCellFactory();
+                            column.setCellFactory(cellFactory);
+                            return column;
+                        })
+                        .collect(Collectors.toList()));
+        tableView.getFixedColumns()
+                .setAll(fixed);
+        data = FXCollections.observableArrayList();
+        tableView.setItems(data);
+        numberRowChildrenModifications = new AtomicLong();
+    }
+
+    private void fillTableData() {
+        data.setAll(IntStream.range(0, NUMBER_OF_ROWS)
+                .mapToObj((int rowIndex) -> new RowItem(rowIndex, NUMBER_OF_COLUMNS))
+                .collect(Collectors.toList()));
+    }
+
+    private void setupManipulators() {
+        dataManipulationCountdown = new CountDownLatch(1);
+        allowedToRun = new AtomicBoolean(true);
+        for (int i = 0; i < THREADS; i++) {
+            this.dataManipulators.submit(() -> {
+                Random random = new SecureRandom();
+
+                try {
+                    dataManipulationCountdown.await();
+                } catch (InterruptedException e) {
+                    // return silently
+                    return;
+                }
+                while (allowedToRun.get()) {
+                    RowItem rowItem = tableView.getItems()
+                            .get(random.nextInt(NUMBER_OF_ROWS));
+                    rowItem.values[random.nextInt(rowItem.values.length)].set((100 + random.nextInt(100)) + "");
+                }
+            });
+        }
+    }
+
+    @After
+    public void afterEach() throws TimeoutException {
+        allowedToRun.set(false);
+        dataManipulators.shutdownNow();
+        FxToolkit.cleanupStages();
+    }
+
+    @Test
+    public void shouldNotFreeze_When_SelectionIsChangedUnderLoad() {
+        fillTableData();
+        produceUpdateLoad();
+
+        Duration timeTaken = measure(this::changeSelection);
+        assertThat(timeTaken, is(lessThan(MAX_DURATION_FOR_SELECTION_CHANGES)));
+    }
+
+    @Test
+    public void shouldNotFreeze_When_ScrollingVerticallyUnderLoad() {
+        fillTableData();
+        produceUpdateLoad();
+
+        Duration timeTaken = measure(this::scrollDownAndUp);
+        assertThat(timeTaken, is(lessThan(MAX_DURATION_FOR_SELECTION_CHANGES)));
+    }
+
+    @Test
+    public void shouldReuseItems_When_ScrollingVertically() {
+        fillTableData();
+        measure(() -> scrollDown(60));
+        numberRowChildrenModifications.set(0);
+        measure(() -> scrollUp(60));
+        assertThat(numberRowChildrenModifications.get(), is(0L));
+    }
+
+    @Test
+    public void shouldHaveNoArtifacts_When_ScrollingVertically() {
```

Comment:
This test passes even when the fix is not applied. Not sure if its supposed to.

Reply:
Yes, it is supposed to be that way. I played around with the code and I thought this test could at least guarantee that the scrolling is not broken. I think it does not harm to have such tests, as they make the life easier when modifying the code